test/unit/dialects/mysql/query-generator.test.js

```diff
@@ -700,6 +700,54 @@ if (dialect === 'mysql') {
             '`Project`',
           ],
           expectation: 'SELECT * FROM `Project`',
+        }, {
```

Comment:
This test file is (slowly) being phased out, we run all tests against all dialects now

Could you remove these tests and add them to https://github.com/sequelize/sequelize/blob/main/test/unit/query-generator/select-query.test.ts instead?

Some dialects will throw but that should be tested too. Here is how we do it for another queryGenerator method: https://github.com/sequelize/sequelize/blob/main/test/unit/query-generator/create-database-query.test.ts#L80-L85

Reply:
I wrote the test code against all dialects(include throw when unsupported option.

In mssql case, they overwrite the abstract query generator's `selectFromTableFragment` method. So I added condition statement in mssql query generator's `selectFromTableFragment` method to make pass the test.

But I wonder is this correct way. Is there any option better this way?

Reply:
That's a good solution :) We have a function called `rejectInvalidOptions` that we typically use for this but right now this is fine